src/config/strgen.rs

```diff
@@ -277,6 +304,33 @@ mod tests {
     fn basic_match() {
         results_in(
             // Equivalent to `d{ incorrect-os: g, default: e }f`.
+            Spec {
+                string: Some("d".to_owned()),
+                spectype: SpecType::match_expr(
+                    vec![
+                        (
+                            Expr {
+                                exprtype: if cfg!(windows) {
+                                    ExprType::Linux
+                                } else {
+                                    ExprType::Windows
+                                },
+                            },
```

Comment:
Determining an incorrect-os is repeated elsewhere in the unit tests. It would be better to evaluate an incorrect-os once and store it for use in more than one unit test function.